src/fortran/cantera_thermo.f90

```diff
@@ -431,11 +431,47 @@ subroutine ctthermo_setState_SP(self, s, p)
       self%err = th_set_sp(self%thermo_id, s, p)
     end subroutine ctthermo_setstate_sp
 
-    subroutine ctthermo_equilibrate(self, XY)
+    subroutine ctthermo_equilibrate(self, XY, solver, rtol, max_steps, max_iter, estimate_equil, log_level)
       implicit none
       type(phase_t), intent(inout) :: self
       character*(*), intent(in) :: XY
-      self%err = th_equil(self%thermo_id, XY)
+      character*(*), intent(in), optional :: solver
+      double precision, intent(in), optional :: rtol
+      integer, intent(in), optional :: max_steps
+      integer, intent(in), optional :: max_iter
+      integer, intent(in), optional :: estimate_equil
+      integer, intent(in), optional :: log_level
+      character*(10) :: solver_
```

Comment:
One of the equilibrium solver names is `element_potential`, so I think this string should be at least large enough to accomodate that?